From 640041c63a59e57c1ccaaadc79925c833ffbb5dd Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Date: Thu, 29 May 2014 10:37:59 +0100
Subject: [PATCH 215/351] switch: arizona: Add option to select headphone
 detection channel

Add a pdata/DT option to select whether to use the left or the right
channel for headphone detection.

Change-Id: I608d3bf29af3491692759099805990696203ffa0
Signed-off-by: Charles Keepax <ckeepax@opensource.wolfsonmicro.com>
Signed-off-by: Nikesh Oswal <Nikesh.Oswal@wolfsonmicro.com>
---
 drivers/switch/switch-arizona.c   |    7 +++++--
 include/linux/mfd/arizona/pdata.h |    6 ++++++
 2 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/drivers/switch/switch-arizona.c b/drivers/switch/switch-arizona.c
index 4c20bea..35935a5 100644
--- a/drivers/switch/switch-arizona.c
+++ b/drivers/switch/switch-arizona.c
@@ -1042,7 +1042,7 @@ int arizona_micd_mic_reading(struct arizona_extcon_info *info, int val)
 done:
 	pm_runtime_mark_last_busy(info->dev);
 
-	if (arizona->pdata.hpdet_moisture_imp)
+	if (arizona->pdata.hpdet_channel)
 		ret = arizona_jds_set_state(info, &arizona_hpdet_right);
 	else
 		ret = arizona_jds_set_state(info, &arizona_hpdet_left);
@@ -1073,7 +1073,7 @@ void arizona_micd_mic_timeout(struct arizona_extcon_info *info)
 
 	dev_dbg(info->arizona->dev, "MICD timed out, reporting HP\n");
 
-	if (arizona->pdata.hpdet_moisture_imp)
+	if (arizona->pdata.hpdet_channel)
 		ret = arizona_jds_set_state(info, &arizona_hpdet_right);
 	else
 		ret = arizona_jds_set_state(info, &arizona_hpdet_left);
@@ -1620,6 +1620,9 @@ static int arizona_extcon_get_pdata(struct arizona *arizona)
 	arizona_of_read_u32(arizona, "wlf,hpdet-moisture-imp", false,
 			    &pdata->hpdet_moisture_imp);
 
+	arizona_of_read_u32(arizona, "wlf,hpdet-channel", false,
+			    &pdata->hpdet_channel);
+
 	return 0;
 }
 #else
diff --git a/include/linux/mfd/arizona/pdata.h b/include/linux/mfd/arizona/pdata.h
index 981bd0c..1d99f74 100644
--- a/include/linux/mfd/arizona/pdata.h
+++ b/include/linux/mfd/arizona/pdata.h
@@ -158,6 +158,12 @@ struct arizona_pdata {
 	 */
 	int hpdet_moisture_imp;
 
+	/**
+	 * Channel to use for headphone detection, valid values are 0 for
+	 * left and 1 for right
+	 */
+	int hpdet_channel;
+
 	/** Use software comparison to determine mic presence */
 	bool micd_software_compare;
 
-- 
1.7.9.5

